### PR TITLE
make presets great again

### LINKED ISSUE #1157: [BUG] Problems Panel is unresponsive after opening the scratch options.
Problems Panel is unresponsive after opening the scratch options.

1. Open a scratch
2. Go to the options panel. The page seems to reload when opening it.
3. Attempt to click on the Problems handle.
4. The problems panel will not respond.

The problems panel should respond when clicking it.

https://cdn.discordapp.com/attachments/897066363951128590/1220738748162117703/20240322-1419-01.8846801.mp4?ex=6610088d&is=65fd938d&hm=75cb7d2e137251cfba8ce264eb110bdd21db7c56f64257736afb05bbde8c19e0&

https://cdn.discordapp.com/attachments/897066363951128590/1220769113262260234/20240322-1617-52.9619414.mp4?ex=661024d5&is=65fdafd5&hm=ca30d0d35d7d566aa81bd37702cb7aaa25f5a3a81bc11124b55a5b1826a36200&

https://decomp.me/scratch/PdEX4

 - OS: Windows
 - Browser Edge

 - Device: iPhone 13
 - OS: iOS 16.3.1
 - Browser Safari

### PR DESCRIPTION
**Pull request by mkst**
- (per ConorB) don't wrap the Options tab in a function to fix the "Problems" issue
- Differentiate between no presets (i.e. either Options tab, or API hasn't returned yet) and empty presets (i.e. PSP platform)
  - show "Loading..." whilst waiting for API to return
- Allow user to select "Custom" preset
- Sort presets alphabetically by name

Closes #1157.

### CODE REVIEW

File: frontend/src/components/compiler/PresetSelect.tsx [hunk 1]
```diff
@@ -21,23 +21,31 @@ export default function PresetSelect({ className, platform, presetId, setPreset,
...
         className={className}
-        options={presetsToOptions(serverPresets)}
+        options={presetsToOptions(sortedPresets)}
         value={selectedPreset?.name || "Custom"}
         onChange={name => {
-            const p = serverPresets.find(p => p.name === name)
-            if (p) {
-                setPreset(p)
-            }
+            setPreset(name === "Custom" ? null : sortedPresets.find((p: api.Preset) => p.name === name))
```
Comment (ethteck) [inline frontend/src/components/compiler/PresetSelect.tsx:48]: Passing null into setPreset was the original issue (the setPreset method was trying to access fields of null and erroring)

Comment (mkst) [reply frontend/src/components/compiler/PresetSelect.tsx:48]: Can you demonstrate this on the vercel deployment? I'll take another look but I tested changing the preset to Custom & saving and everything behaved as expected

Comment (ethteck) [reply frontend/src/components/compiler/PresetSelect.tsx:48]: If I go to the new scratch page and toggle between something and custom, I get

image

Comment (mkst) [reply frontend/src/components/compiler/PresetSelect.tsx:48]: Aha. I've been testing the "Custom" on the Scratch Options tab which is where it works. Let me figure out a fix for the New Scratch page 👍

**Review by ethteck — APPROVED** [review]

**Pull request merged**